File: src/test/java/com/raychenon/leetcode/FindTheDuplicateNumberTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 28/3/2020
 * https://leetcode.com/problems/find-the-duplicate-number/
 */
public class FindTheDuplicateNumberTest {

    private static FindTheDuplicateNumber instance;

    @BeforeClass
    public static void setUp() {
        instance = new FindTheDuplicateNumber();
    }

    @Test
    public void example1Test() {
        int[] array = {1, 3, 4, 2, 2};
        assertMethods(2, array);
    }

    @Test
    public void example2Test() {
        int[] array = {3, 1, 3, 4, 2};
        assertMethods(3, array);
    }

    @Test
    public void noDuplicateTest() {
        int[] array = {3};
        assertMethods(-1, array);
    }

    public void assertMethods(int expectedDuplicate, int[] array) {
        assertEquals(expectedDuplicate, instance.findDuplicate(array));
        assertEquals(expectedDuplicate, instance.findDuplicateMemory(array));
    }
}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/FindTheDuplicateNumberTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.TestCase
import org.junit.Test

/**
 * User: raychenon
 * Date: 29/3/2020
 * https://leetcode.com/problems/find-the-duplicate-number/
 */
class FindTheDuplicateNumberTest {

    @Test
    fun example1Test() {
        val array = intArrayOf(1, 3, 4, 2, 2)
        assertMethods(2, array)
    }

    @Test
    fun example2Test() {
        val array = intArrayOf(3, 1, 3, 4, 2)
        assertMethods(3, array)
    }


    fun assertMethods(expectedDuplicate: Int, array: IntArray) {
        TestCase.assertEquals(
            expectedDuplicate,
            FindTheDuplicateNumber.findDuplicate(array)
        )
        TestCase.assertEquals(
            expectedDuplicate,
            FindTheDuplicateNumber.findDuplicateMemory(array)
        )
    }


}